File: backend/models/practitioner.py
"""
Practitioner models for the SANA Algorithms Suite.

Contains Pydantic models for CAM practitioner data.
"""

from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from uuid import UUID, uuid4


class Practitioner(BaseModel):
    """Base practitioner model."""
    id: UUID = Field(default_factory=uuid4)
    full_name: str
    specialties: List[str]
    credentials: List[str]
    verified: bool = False
    sana_index_score: Optional[float] = None
    hourly_rate: Optional[float] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)


class PractitionerProfile(BaseModel):
    """Extended practitioner profile with additional details."""
    practitioner_id: UUID
    bio: Optional[str] = None
    years_experience: Optional[int] = None
    location: Optional[str] = None
    availability: dict = {}
    modalities: List[str] = []
    languages: List[str] = ["English"]


class PractitionerCreate(BaseModel):
    """Model for practitioner creation requests."""
    full_name: str
    specialties: List[str]
    credentials: List[str]
    hourly_rate: Optional[float] = None


class PractitionerResponse(BaseModel):
    """Model for practitioner API responses."""
    id: UUID
    full_name: str
    specialties: List[str]
    credentials: List[str]
    verified: bool
    sana_index_score: Optional[float]
    hourly_rate: Optional[float]

    class Config:
        from_attributes = True


class PractitionerMatch(BaseModel):
    """Model for practitioner matching results."""
    practitioner: Practitioner
    match_score: float
    match_reasons: List[str]

File: frontend/lib/models/practitioner.dart
class Practitioner {
  final String id;
  final String fullName;
  final List<String> specialties;
  final List<String> credentials;
  final bool verified;
  final double? sanaIndexScore;
  final double? hourlyRate;

  Practitioner({
    required this.id,
    required this.fullName,
    required this.specialties,
    required this.credentials,
    this.verified = false,
    this.sanaIndexScore,
    this.hourlyRate,
  });

  factory Practitioner.fromJson(Map<String, dynamic> json) {
    return Practitioner(
      id: json['id'],
      fullName: json['full_name'],
      specialties: List<String>.from(json['specialties']),
      credentials: List<String>.from(json['credentials']),
      verified: json['verified'] ?? false,
      sanaIndexScore: json['sana_index_score']?.toDouble(),
      hourlyRate: json['hourly_rate']?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'full_name': fullName,
      'specialties': specialties,
      'credentials': credentials,
      'verified': verified,
      'sana_index_score': sanaIndexScore,
      'hourly_rate': hourlyRate,
    };
  }
}
